conjur/logic/hostfactory_logic.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+"""
+HostFactoryLogic
+
+This module holds the business logic for executing and manipulating
+returned data
+"""
+
+# Builtins
+import json
+from urllib import parse
+
+# Internals
+from conjur.data_object.create_token_data import CreateTokenData
+from conjur.errors import MissingRequiredParameterException
+
+
+# pylint: disable=too-few-public-methods
+class HostFactoryLogic:
+    """
+    HostFactoryLogic
+
+    This class holds the business logic for executing and manipulating
+    returned data
+    """
+
+    def __init__(self, client):
+        self.client = client
+
+    def create_token(self, create_token_data: CreateTokenData) -> str:
+        """
+        Creates a host factory token using the parameters in the 'create_token_data' argument.
+        Returns the generated token.
+        """
+        if create_token_data is None:
+            raise MissingRequiredParameterException('create_token_data cannot be empty!')
+
+        # parse.urlencode, If any values in the query arg are sequences and doseq is true, each
+        # sequence element is converted to a separate parameter.
+        # This is set to True to handle CreateTokenData.cidr which is a list
+        response = self.client.create_token(parse.urlencode(create_token_data.to_dict(),
+                                                            doseq=True))
+        if response is not None:
+            data = response.json()
+            if data is not None and len(data) > 0:
```

Comment:
why do we need a compound check?
if Data is not none then its length automatically will be greater > 0 no?

Reply:
not necessarily